lib/connections/limiter.go

```diff
@@ -110,33 +110,36 @@ func (lim *limiter) processDevicesConfigurationLocked(from, to config.Configurat
 	}
 
 	// Delete remote devices which were removed in new configuration
-	for _, dev := range from.Devices {
-		if _, ok := seen[dev.DeviceID]; !ok {
-			l.Debugf("deviceID: %s should be removed", dev.DeviceID)
+	for devID := range lim.deviceWriteLimiters {
+		if _, ok := seen[devID]; !ok {
+			l.Debugf("deviceID: %s should be removed", devID)
 
-			delete(lim.deviceWriteLimiters, dev.DeviceID)
-			delete(lim.deviceReadLimiters, dev.DeviceID)
+			delete(lim.deviceWriteLimiters, devID)
+			delete(lim.deviceReadLimiters, devID)
 		}
 	}
 }
 
-func (lim *limiter) VerifyConfiguration(from, to config.Configuration) error {
+func (lim *limiter) VerifyConfiguration(to config.Configuration) error {
 	return nil
 }
 
-func (lim *limiter) CommitConfiguration(from, to config.Configuration) bool {
+func (lim *limiter) CommitConfiguration(to config.Configuration) bool {
 	// to ensure atomic update of configuration
 	lim.mu.Lock()
 	defer lim.mu.Unlock()
 
 	// Delete, add or update limiters for devices
-	lim.processDevicesConfigurationLocked(from, to)
+	lim.processDevicesConfigurationLocked(to)
 
-	if from.Options.MaxRecvKbps == to.Options.MaxRecvKbps &&
-		from.Options.MaxSendKbps == to.Options.MaxSendKbps &&
-		from.Options.LimitBandwidthInLan == to.Options.LimitBandwidthInLan {
+	if lim.firstConf && lim.opts.MaxRecvKbps == to.Options.MaxRecvKbps &&
```

Comment:
`firstConf` is badly named -- it's false for the first config. I stared a long time at this `if` wondering why you would ignore the first conf change only.